### PR TITLE
USHIFT-4932: TLS config options for api server

### PR DESCRIPTION
**Pull request by pacevedom**
**Which issue(s) this PR addresses**:

Closes #<Issue Number>

### CODE REVIEW

**Review by pmtk — COMMENTED** [review]
What about `RouteControllerManager` and `ClusterPolicyController` (there might be other)?
`openshiftcontrolplanev1.OpenShiftControllerManagerConfig` -> `configv1.HTTPServingInfo` -> `configv1.ServingInfo`:
```
type ServingInfo struct {
	...

	// MinTLSVersion is the minimum TLS version supported.
	// Values must match version names from https://golang.org/pkg/crypto/tls/#pkg-constants
	MinTLSVersion string `json:"minTLSVersion,omitempty"`
	// CipherSuites contains an overridden list of ciphers for the server to support.
	// Values must match cipher suite IDs from https://golang.org/pkg/crypto/tls/#pkg-constants
	CipherSuites []string `json:"cipherSuites,omitempty"`
}
```

File: pkg/config/apiserver.go [hunk 1]
```diff
@@ -55,3 +65,58 @@ type AuditLog struct {
...
 }
+
+type TLSConfig struct {
+	// CipherSuites lists the allowed cipher suites that the API server will
+	// accept and serve.
+	CipherSuites []string `json:"cipherSuites"`
+
+	// MinVersion specifies which TLS version is the minimum version of TLS
+	// to serve from the API server.
+	// +kubebuilder:validation:Enum:=VersionTLS12;VersionTLS13
```
[the remark this answers is not in the document]
Comment (pacevedom) [reply pkg/config/apiserver.go:78]: Done

**Comment by pacevedom** [comment]
> What about `RouteControllerManager` and `ClusterPolicyController` (there might be other)? `openshiftcontrolplanev1.OpenShiftControllerManagerConfig` -> `configv1.HTTPServingInfo` -> `configv1.ServingInfo`:
> 
> ```
> type ServingInfo struct {
> 	...
> 
> 	// MinTLSVersion is the minimum TLS version supported.
> 	// Values must match version names from https://golang.org/pkg/crypto/tls/#pkg-constants
> 	MinTLSVersion string `json:"minTLSVersion,omitempty"`
> 	// CipherSuites contains an overridden list of ciphers for the server to support.
> 	// Values must match cipher suite IDs from https://golang.org/pkg/crypto/tls/#pkg-constants
> 	CipherSuites []string `json:"cipherSuites,omitempty"`
> }
> ```

Added `RouteControllerManager`, as `ClusterPolicyController` does not expose and endpoint where it listens. Good catch!

File: pkg/config/apiserver.go [hunk 2]
```diff
@@ -55,3 +65,60 @@ type AuditLog struct {
...
+		// Golang will ignore whatever is configured.
+		t.CipherSuites = getIANACipherSuites(configv1.TLSProfiles[configv1.TLSProfileModernType].Ciphers)
+	}
+}
+
+func (t *TLSConfig) Validate() error {
+	if len(t.CipherSuites) == 0 {
+		return fmt.Errorf("unsupported empty cipher suites")
+	}
+	var cipherSuites []string
```
Comment (pmtk) [inline pkg/config/apiserver.go:105]: This might make it more obvious
```suggestion
	var allowedCipherSuites []string
```

Comment (pacevedom) [reply pkg/config/apiserver.go:105]: Was keeping the syntax from other control plane components, such as kubelet or apiserver. All of them use `--tls-cipher-suites` and `-tls-min-version`. The ingress operator [also uses](https://github.com/openshift/cluster-ingress-operator/blob/8be174945db79b0660bd0fd21ed279ab7b6aafbf/manifests/02-deployment.yaml#L98-L100) this syntax. Should we change it to make it different?

**Pull request merged**